lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReader.java

```diff
@@ -13,13 +13,13 @@
  */
```

Comment:
What's the "simplification" in this commit? It looks equally complex as before.

Reply:
I'm updating the code a bit more now to improve it's readability.
To my eyes the resulting code is simpler because we now avoid methods with 3 or 4 levels of `if`, remove unnecessary class member variables, add `@Nullable` annotation in a couple of places and remove weird usage of List in `listWithNulls`.